go/ai/request_helpers.go

```diff
@@ -66,3 +66,55 @@ func NewTextMessage(role Role, text string) *Message {
 		Content: []*Part{NewTextPart(text)},
 	}
 }
+
+// WithCacheTTL adds cache TTL configuration for the desired message
+func (m *Message) WithCacheTTL(ttlSeconds int) *Message {
+	metadata := make(map[string]any)
+
+	if m.Metadata != nil {
+		for k, v := range m.Metadata {
+			metadata[k] = v
+		}
+	}
+
+	cache := make(map[string]any)
+	cache["ttlSeconds"] = ttlSeconds
+
+	if existingCache, ok := metadata["cache"].(map[string]any); ok {
+		existingCache["ttlSeconds"] = ttlSeconds
+	} else {
+		metadata["cache"] = cache
+	}
```

Comment:
I think this can be replaced with `maps.Copy()` like here: https://github.com/firebase/genkit/blob/e373b699c6ae72c01daa5e349d8b807d9ae9052c/go/ai/prompt.go#L72